packages/rspack/src/builtin-plugin/css-extract/hmr/hotModuleReplacement.ts

```diff
@@ -5,53 +5,48 @@
   func-names
 */
 
-/** @typedef {any} TODO */
+const normalizeUrlFn = require("./normalize-url");
```

Comment:
nice, but if only it could be converted into esm 

`normalizeUrl` is a relatively short function, and I think it is acceptable if it can be written in this file, If it can't be converted into esm, we can try this.